src/Stryker.Core/Stryker.Core/Initialisation/Buildalyzer/IAnalyzerResultExtensions.cs

```diff
@@ -103,6 +103,32 @@ public static string GetSymbolFileName(this IAnalyzerResult analyzerResult)
             return Path.ChangeExtension(analyzerResult.GetAssemblyName(), ".pdb");
         }
 
+        public static IEnumerable<ISourceGenerator> GetSourceGenerators(this IAnalyzerResult analyzerResult)
+        {
+            var generators = new List<ISourceGenerator>();
+            foreach (var analyzer in analyzerResult.AnalyzerReferences)
+            {
+                try
+                {
+                    var assembly = System.Reflection.Assembly.LoadFile(analyzer);
+                    foreach (var type in assembly.ExportedTypes)
+                    {
+                        if (type.GetInterface("ISourceGenerator") != null)
+                        {
+                            var generator = type.GetConstructor(Type.EmptyTypes).Invoke(null);
+                            generators.Add(generator as ISourceGenerator);
+                        }
+                    }
+                }
+                catch
+                {
+                    // many framework analyzers won't be loadable
```

Comment:
Reasonable request. Considering the average case - an analyzer that is a package which does not generate code - would you object to logging this issue at the Debug level?

Reply:
not sure:
1. This problem could happen with code generator, then this is obviously a blocking issue
2. We need to have support for analyzer as analyzer may kill mutations through static analysis (we have a request for this feature)
This imply INFO for analyzer and probably Error for source code generator, assuming you can distinguish those

Reply:
Gotcha. Right now I'm loading the assembly to determine if it's a source generator so I can't distinguish up front. It might be spammy but I'll make it a warning.